File: app/src/main/java/com/jcons/pdfconverter/pdfconverter/Trailer.java
package com.jcons.pdfconverter.pdfconverter;


public class Trailer extends Base {

    private int mXRefByteOffset;
    private int mObjectsCount;
    private String mId;
    private Dictionary mTrailerDictionary;

    public Trailer() {
        clear();
    }

    public void setId(String value) {
        mId = value;
    }

    public void setCrossReferenceTableByteOffset(int value) {
        mXRefByteOffset = value;
    }

    public void setObjectsCount(int value) {
        mObjectsCount = value;
    }

    private void renderDictionary() {
        mTrailerDictionary.setContent("  /Size " + Integer.toString(mObjectsCount));
        mTrailerDictionary.addNewLine();
        mTrailerDictionary.addContent("  /Root 1 0 R");
        mTrailerDictionary.addNewLine();
        mTrailerDictionary.addContent("  /ID [<" + mId + "> <" + mId + ">]");
        mTrailerDictionary.addNewLine();
    }

    private String render() {
        renderDictionary();
        StringBuilder sb = new StringBuilder();
        sb.append("trailer");
        sb.append("\n");
        sb.append(mTrailerDictionary.toPDFString());
        sb.append("startxref");
        sb.append("\n");
        sb.append(mXRefByteOffset);
        sb.append("\n");
        sb.append("%%EOF");
        sb.append("\n");
        return sb.toString();
    }

    @Override
    public String toPDFString() {
        return render();
    }

    @Override
    public void clear() {
        mXRefByteOffset = 0;
        mTrailerDictionary = new Dictionary();
    }

}

File: app/src/main/java/com/jcons/pdfconverter/pdfconverter/Trailer.kt
package com.jcons.pdfconverter.pdfconverter


class Trailer : Base {
    private var mXRefByteOffset = 0
    private var mObjectsCount = 0
    private var mId: String? = null
    private var mTrailerDictionary: Dictionary? = null
    fun setId(value: String?) {
        mId = value
    }

    fun setCrossReferenceTableByteOffset(value: Int) {
        mXRefByteOffset = value
    }

    fun setObjectsCount(value: Int) {
        mObjectsCount = value
    }

    private fun renderDictionary() {
        mTrailerDictionary!!.content = "  /Size " + Integer.toString(mObjectsCount)
        mTrailerDictionary!!.addNewLine()
        mTrailerDictionary!!.addContent("  /Root 1 0 R")
        mTrailerDictionary!!.addNewLine()
        mTrailerDictionary!!.addContent("  /ID [<$mId> <$mId>]")
        mTrailerDictionary!!.addNewLine()
    }

    override fun toPDFString(): String {
        renderDictionary()
        val sb = StringBuilder()
        sb.append("trailer")
        sb.append("\n")
        sb.append(mTrailerDictionary!!.toPDFString())
        sb.append("startxref")
        sb.append("\n")
        sb.append(mXRefByteOffset)
        sb.append("\n")
        sb.append("%%EOF")
        sb.append("\n")
        return sb.toString()
    }

    override fun clear() {
        mXRefByteOffset = 0
        mTrailerDictionary = Dictionary()
    }

    init {
        clear()
    }
}
